### ISSUE TITLE
How do i add a new system like the wii u

### ISSUE THREAD

**Issue by Aaskari1988** [Aloshi/EmulationStation#571]
??

**Comment by mikekornikh** [comment]
The "systems" in ES are just pre-made folders with theme images, etc. You can define a platform in the es_system.cfg file by just naming it wiiu. You can set the theme to just wii because they look pretty much the same lol.

Try something like this
<system>
<name>wiiu</name>
<fullname>Wii U</fullname>
<path>~/.emulationstation/roms/nameofwiiupath</path>
<extension>.lnk</extension>
<command>**emulator exe %ROM%</command>
<platform>wiiu</platform>
<theme>wii</theme>
</system>

**Comment by Aaskari1988** [comment]
Thanks gonna try

**Comment by Aaskari1988** [comment]
it worked very nice. now i just need to create the text svg or png for the logo do you know the dimentions for that?

**Comment by mikekornikh** [comment]
Basically, you should just copy the wii folder from the themes folder and rename it "wiiu". Get a .png icon from an image search and place it in that folder. From there, open the .xml file in the wiiu folder and change all the directories to the new wiiu folder you created, the image refrence from the old wii .svg to the new .png, and name to "Nintendo Wii U" from "Nintendo Wii"

**Comment by Aaskari1988** [comment]
Got it working.  The images I found doesn't exactly look great but fine temporarily.

**Comment by Alobex** [comment]
Whoah. How did you get a Wii U Rom?

**Comment by Aaskari1988** [comment]
Emulator http://cemu.info/ still early it dont really play anything perfectly but its moving pretty fast.
games not hard to find.

**Comment by Alobex** [comment]
Yeah, but how do you get the wii u roms. Do you use a program to extract the iso from the disk?

**Comment by Aaskari1988** [comment]
I downloaded the games online.  Just the once I have on disk of course.

**Comment by Aaskari1988** [comment]
The games are WUD format.

**Comment by Alobex** [comment]
Ok, but which website?

**Comment by dankcushions** [comment]
can we not use github to exchange links to illegal software, please.

http://lmgtfy.com/?q=wii+u+roms

**Comment by Alobex** [comment]
Oh Yeah, I forgot. The legal way to do this is to find a software that can extract the iso from the wii u disk, not a wii disk.

**Comment by Lowitle** [comment]
I tried to configure my emulationstation to setup cemu emulator. I have no success. I have this config and cemu starts, but dont launch any game.


```	
	<system>
		<name>wiiu</name>
		<fullname>WiiU</fullname>
		<path>~\.emulationstation\roms\wiiu</path>
		<extension>.wud</extension>
		<command>%HOMEPATH%\.emulationstation\systems\cemu\cemu.exe "%ROM_RAW%"</command>
		<platform>wiiu</platform>
		<theme>wiiu</theme>	
	</system>
```

Someone knows how to make it work correctly?

**Comment by IncognitoJam** [comment]
@Lowitle I also had this issue. This is what I used:
```
	<system>
		<name>wiiu</name>
		<fullname>WiiU</fullname>
		<path>~\.emulationstation\roms\wiiu</path>
		<extension>.wud</extension>
		<command>%HOMEPATH%\.emulationstation\systems\cemu\cemu.exe -f -g "%ROM_RAW%"</command>
		<platform>wiiu</platform>
		<theme>wiiu</theme>	
	</system>
```

The `-f` tells CEMU to launch the game in full screen mode, and the `-g` allows you to provide a game file to launch. [Source](https://www.reddit.com/r/cemu/comments/507qce/current_launch_commands/)

**Comment by Tchennn** [comment]
Hello and thanks for all the very usefull information.
Howerver, I have used the IncognitoJam script but I get some messages of error "Unable to create file keys.txt" (twice) when I try to launch a WiiU game through EmulationStation. Do you have, guys, any idea?
Thx for your help. If I close the error windows it works at the end but that 's not very " esthetic"....

**Comment by korreckj328** [comment]
Tchennn I have the same problem plus it won't recognize a controller when launched from emulationstation.  Not sure what the deal is, as I used to be able to launch cemu this way about six months ago.

**Comment by Tchennn** [comment]
Hi korreckj328,
I have found a solution, but not so academic....
First I have created .bat files in order to launch each of my wiiu roms, then in a new folder I have created shortcuts for each of the .bat files. Finally, I have modified EmulationStation config file to launch the roms from the shortcuts... and it works!!!

**Comment by MrTom4hawk** [comment]
Hi @Tchennn .
I was wondering if u can give me an example of any of ur bat files. I tried to make one but cemu gave me an error.
This is my bat file: 
"C:\Portable_Game_Station_1.5\.emulationstation\systems\cemu\Cemu.exe" -g "C:\Portable_Game_Station_1.5\.emulationstation\roms\wiiu\mariokart8\code\Turbo.rpx" -f"
exit
it does not work. 
in my system config file i have the one that IncognitoJam posted. I've only change the extension (.wud .WUD to  .bat .BAT).
Can u please help me?.
Sorry for my english !
Take care!

**Comment by Tchennn** [comment]
Hi @MrTom4hawk 
@echo off
"C:\Users\Arcade\.emulationstation\systems\cemu_1.11.3\cemu.exe" -g "C:\Users\Arcade\.emulationstation\Roms\wiiu\MKARTVIII\MARIO KART 8 [AMKP0101]\code\Turbo.rpx" -f
exit

too much " in your code!

best

**Comment by BryceGre** [comment]
If I may, I found a better solution.

The problem is that cemu.exe looks in the working directory for it data files (keys.txt, the controllerProfiles folder, etc.) and if it is launched from a directory that is not it's folder (such as the directory containing EmulationStation) it won't be able to find the files.

The solution (until EmluationStation is updated to use the working directory of it's target) is to replace:

`<command>C:\Users\{username}\.emulationstation\systems\cemu\cemu.exe -f -g "%ROM_RAW%"</command>`

with:

`<command>START /D C:\Users\{username}\.emulationstation\systems\cemu\ cemu.exe -f -g "%ROM_RAW%"</command>`

NOTE THE SPACE between the directory and cemu.exe. Here, the START command runs "cemu.exe" using the working directory (specified by the flag /D) of "C:\Users\\{username}\\.emulationstation\systems\cemu\". Make sure to set the working directory to the directory containing cemu.exe, then not only can we launch cemu.exe without it's full path but it will look in it's own directory for it's required files.

**Comment by MrTom4hawk** [comment]
@BryceGre @Tchennn  Hi!, thx for the help u guys!.
I'm not at home right now. I'm gonna try to do this on monday.
I have 1 question.

Here is my cemu folder exact location.
D:\Portable_Game_Station_1.5\ .emulationstation\systems\cemu\Cemu.exe

So... the systems.cfg file goes like this:

<command>START /D D:\Portable_Game_Station_1.5\.emulationstation\systems\cemu\ cemu.exe -f -g "%ROM_RAW%"</command>   

is that correct??... and with the space between the directory and cemu.exe too?

Thank u very much for everything!

**Comment by BryceGre** [comment]
@MrTom4hawk Yes that's correct, assuming that you meant to have a backslash between "Portable_Game_Station_1.5" and ".emulationstation" but GitHub is hiding it. (it did that to me too for some reason)
With the backslash it should look like this:
`START /D D:\Portable_Game_Station_1.5\.emulationstation\systems\cemu\ Cemu.exe -f -g "%ROM_RAW%"`

**Comment by MrTom4hawk** [comment]
@BryceGrees yes, for some reason github is hiding the backlash.
Ok!. Thx so much!.

**Comment by xev1n** [comment]
Thank you for finding the START /D option. A couple questions. 1. It does not seem to work with %HOME%\.emulationstation\systems\ I have to actually specify a path which works fine. Any idea on how to use %HOME% so it isn't locked to a specific drive letter? 2. How do you run loadline games thru ES? I can get it to work fine with wud and wux but I think that's because they are only one file. Thanks for your help.

**Comment by MrTom4hawk** [comment]
Hi! @xev1n 

This is my conf system file.

<system>
<name>wiiu</name>
<fullname>WiiU</fullname>
<path>~\.emulationstation\roms\wiiu</path>
<extension>.rpx .RPX</extension>
<command>START /D D:\Portable_Game_Station_1.5\.emulationstation\systems\cemu\ Cemu.exe -f -g "%ROM_RAW%"</command>
<platform>wiiu</platform>
<theme>wiiu</theme>	
</system>

Works like a charm. Thx to @BryceGre by the way.

**Comment by xev1n** [comment]
@BryceGre Yes Windows. Thanks for this I will try it out. Do you know if I can specify multiple rom paths in the systems.cfg file. My thinking is I can just path to each and every Wii U game that uses .rpx files/loadiine games. I am going to try this out on my own as well, was just wondering if you had any experience with this? Thanks again!

**Comment by BryceGre** [comment]
@xev1n I took a look at the code and, unfortunately, it doesn't support multiple paths for roms. It's a good idea though.
